src/unix.rs

```diff
@@ -42,6 +45,17 @@ fn dup_fd(fd: RawFd) -> io::Result<Stdio> {
     dup_result.map(File::into_stdio)
 }
 
+fn nix_err_to_io_err(err: nix::Error) -> io::Error {
+    match err {
+        Sys(err_no) => {
+            io::Error::from(err_no)
+        }
+        _ => {
+            io::Error::new(ErrorKind::InvalidData, err)
+        }
+    }
+}
```

Comment:
Thanks for the PR! Sorry for taking a while to get to it. Seems unfortunate that we need to implement this conversion ourselves -- does nix not provide anything?